tests/end2end/features/test_completion_bdd.py

```diff
@@ -0,0 +1,29 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015-2016 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+import pytest_bdd as bdd
+bdd.scenarios('completion.feature')
+
+
+@bdd.then(bdd.parsers.parse("the completion model should be {model}"))
```

Comment:
Yeah, for inspecting completion content I think the unit tests are your best bet right now. I think this should be unit-tested in `test_update_completion` regardless of whether we have a feature test for it.
A non-flaky end2end completion test, would be a great addition though. Maybe the `:open <Tab> <Tab> <Enter>` approach works fine and I was just doing something stupid before.